features/dd-sdk-android-trace/src/test/kotlin/com/datadog/trace/core/CoreSpanBuilderTest.kt

```diff
@@ -0,0 +1,534 @@
+        val spans = mutableListOf<DDSpan>()
+        val nbSamples = 10
+
+        // root (aka spans[0]) is the parent
+        // others are just for fun
+
+        val root = tracer.buildSpan(instrumentationName, "fake_O").withServiceName("foo").start()
+
+        var lastSpan = root
+
+        for (i in 1..nbSamples) {
+            lastSpan = tracer
+                .buildSpan(instrumentationName, "fake_$i")
+                .withServiceName("foo")
+                .asChildOf(lastSpan.context())
+                .start() as DDSpan
+            spans.add(lastSpan)
+            lastSpan.finish()
+        }
+
+        val trace = root.context().trace as PendingTrace
+        assertThat(trace.rootSpan).isEqualTo(root)
+        assertThat(trace.size()).isEqualTo(nbSamples)
+        assertThat(trace.spans).containsAll(spans)
+        assertThat(spans[(Math.random() * nbSamples).toInt()].context().trace.spans).containsAll(spans)
+    }
+
+    @ParameterizedTest
+    @MethodSource("extractedContextProvider")
+    fun `ExtractedContext should populate new span details`(extractedContext: ExtractedContext) {
+        // setup:
+        val thread = Thread.currentThread()
+        val span = tracer.buildSpan(instrumentationName, "op name")
+            .asChildOf(extractedContext).start() as DDSpan
+
+        // expect:
+        assertThat(span.traceId).isEqualTo(extractedContext.traceId)
+        assertThat(span.parentId).isEqualTo(extractedContext.spanId)
+        assertThat(span.samplingPriority).isEqualTo(extractedContext.samplingPriority)
+        assertThat(span.context().origin).isEqualTo(extractedContext.origin)
+        assertThat(span.context().baggageItems).isEqualTo(extractedContext.baggage)
+        // check the extracted context has been copied into the span tags. Intercepted tags will be skipped from
+        // propagation.
+        extractedContext.tags.filterKeys { it != DDTags.ORIGIN_KEY }.forEach { (key, value) ->
+            assertThat(span.context().tags[key]).isEqualTo(value)
+        }
+        assertThat(span.getTag(DDTags.THREAD_ID)).isEqualTo(thread.id)
+        assertThat(span.getTag(DDTags.THREAD_NAME)).isEqualTo(thread.name)
+        assertThat(span.context().propagationTags.headerValue(PropagationTags.HeaderType.DATADOG)).isEqualTo(
+            extractedContext.propagationTags.headerValue(PropagationTags.HeaderType.DATADOG)
+        )
+    }
+
+    @ParameterizedTest
+    @MethodSource("tagContextProvider")
+    fun `TagContext should populate default span details`(tagContext: TagContext) {
+        // setup:
+        val thread = Thread.currentThread()
+        val span = tracer.buildSpan(instrumentationName, "op name").asChildOf(tagContext).start() as DDSpan
+
+        // expect:
+        assertThat(span.traceId).isNotEqualTo(DDTraceId.ZERO)
+        assertThat(span.parentId).isEqualTo(DDSpanId.ZERO)
+        val samplingPriority = span.samplingPriority
+        assert(samplingPriority == null)
```

Comment:
```suggestion
        assertThat(samplingPriority).isNull()
```

Reply:
yeah for some reason that is not working, failing with a NPE and I think is because of the Java function signature at the end of the line.